src/services/data/data.service.ts

```diff
@@ -7,39 +7,72 @@ import {
   revenueMeasure,
   sourceCategory,
 } from '../../models/data-cube/presets';
+import { DataCube } from '../../models/data-cube/data-cube.model';
 import { betweenDates } from '../../models/data-cube/filters';
 import { generateCube } from 'src/models/data-cube/generation';
+import { Injectable, OnDestroy } from '@angular/core';
+import { map, takeUntil, throttleTime } from 'rxjs/operators';
+import { Subject, BehaviorSubject } from 'rxjs';
+import { PreferenceService } from '../preference/preference.service';
+@Injectable()
+export class DataService implements OnDestroy {
+  private static categories = [countryCategory, browserCategory, sourceCategory];
+  private static measures = [activeUserMeasure, revenueMeasure, eventCountMeasure];
+  private static settings = {
+    avgHits: 10000,
+    hitStdDev: 100,
+    avgUsers: 100,
+    userStdDev: 1,
+    avgSessionsPerUser: 5,
+    sessionsPerUserStdDev: 3
+  };
 
-export class DataService {
-  private dataCube = generateCube(
-    [countryCategory, browserCategory, sourceCategory],
-    [activeUserMeasure, revenueMeasure, eventCountMeasure],
-    {
-      avgHits: 10000,
-      hitStdDev: 100,
-      avgUsers: 100,
-      userStdDev: 1,
-      avgSessionsPerUser: 5,
-      sessionsPerUserStdDev: 3,
-    },
-  );
+  private dataCube$ = new BehaviorSubject<DataCube>(generateCube(
+    DataService.categories,
+    DataService.measures,
+    DataService.settings,
+  ));
+  private destroy$ = new Subject();
+
+  constructor(private preferenceService: PreferenceService) {
+    this.preferenceService.data$
+      .pipe(takeUntil(this.destroy$))
+      .pipe(throttleTime(1000))
```

Comment:
I think `throttleTime` pipe can be modified a bit to remove kinda redundant initialization of `dataCube$` in line 30.
https://stackoverflow.com/questions/37146081/rxjs-throttle-behavior-get-first-value-immediately

In that case, we can also remove `this.settings`.

Reply:
I replaced `BehaviorSubject` with `ReplaySubject` to remove redundant initialization, thanks!

Reply:
I just realized it should be `deboundTime`, not `throttleTime`.

Reply:
`throttleTime` with settings `{ leading: true, trailing: true}` works as expected